tapchannel/aux_leaf_signer_test.go

```diff
@@ -0,0 +1,172 @@
+		Nonce:      nonce,
+	}, nil
+}
+
+func somePartialSigWithNonce(t *testing.T) lnwire.OptPartialSigWithNonceTLV {
+	sig, err := randPartialSigWithNonce()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return tlv.SomeRecordT(tlv.NewRecordT[
+		lnwire.PartialSigWithNonceType,
+		lnwire.PartialSigWithNonce,
+	](*sig))
+}
+
+// TestMaxCommitSigMsgSize attempts to find values for the max number of asset
+// IDs we want to allow per channel and the resulting maximum number of HTLCs
+// that channel can allow. The maximum number of different asset IDs that can be
+// committed to a channel directly impacts the number of HTLCs that can be
+// created on that channel, because we have a limited message size to exchange
+// the second-stage HTLC signatures. The goal of this test is to find the right
+// number of asset IDs we should allow per channel to still give us a reasonable
+// amount of HTLCs.
+func TestMaxCommitSigMsgSize(t *testing.T) {
+	// This test is only relevant once, to find the values we want to use
+	// for the maximum number of asset IDs and the resulting maximum number
+	// of HTLCs. We only need to re-run this if any of the parameters
+	// change.
+	t.Skip("Test for manual execution only")
+
+	const maxNumAssetIDs = 10
+
+	var buf bytes.Buffer
+	for numID := 0; numID <= maxNumAssetIDs; numID++ {
+		for htlcs := startNumHTLCs; htlcs <= endNumHTLCs; htlcs++ {
+			buf.Reset()
+
+			msg := makeCommitSig(t, numID, htlcs)
+			err := msg.Encode(&buf, 0)
+			require.NoError(t, err)
+
+			if buf.Len() > lnwire.MaxMsgBody {
+				t.Logf("Last valid commit sig msg size with: "+
+					"numAssetIDs=%d, numHTLCs=%d",
+					numID, htlcs-1)
+
+				break
+			}
+
+			if htlcs == endNumHTLCs {
+				t.Logf("Last valid commit sig msg size with: "+
+					"numAssetIDs=%d, numHTLCs=%d",
+					numID, htlcs)
+			}
+		}
+	}
+}
+
+func makeCommitSig(t *testing.T, numAssetIDs, numHTLCs int) *lnwire.CommitSig {
+	var (
+		msg = &lnwire.CommitSig{
+			HtlcSigs: make([]lnwire.Sig, numHTLCs),
+		}
+		err error
+	)
+
+	// Static values that are always set for custom channels (which are
+	// Taproot channels, so have an all-zero legacy commit signature and a
+	// partial MuSig2 signature).
+	msg.PartialSig = somePartialSigWithNonce(t)
+	msg.CommitSig, err = lnwire.NewSigFromSignature(testSig)
+	require.NoError(t, err)
+
+	assetSigs := make([][]*cmsg.AssetSig, numHTLCs)
+	for i := range numHTLCs {
+		msg.HtlcSigs[i] = testSchnorrSig
+
+		assetSigs[i] = make([]*cmsg.AssetSig, numAssetIDs)
+		for j := range numAssetIDs {
+			var assetID asset.ID
+
+			_, err := rand.Read(assetID[:])
+			require.NoError(t, err)
+
+			assetSigs[i][j] = cmsg.NewAssetSig(
+				assetID, testSchnorrSig, txscript.SigHashAll,
+			)
+		}
+	}
+
+	if numAssetIDs == 0 {
+		return msg
+	}
+
+	commitSig := cmsg.NewCommitSig(assetSigs)
+
+	var buf bytes.Buffer
+	err = commitSig.Encode(&buf)
+	require.NoError(t, err)
+
+	msg.CustomRecords = lnwire.CustomRecords{
+		65543: buf.Bytes(),
```

Comment:
There must be a const for 65543 somewhere? I couldn't find one. But I think it's the custom records TLV int lower bound.